contracts/implementations/nodes/AftermarketDevice.sol

```diff
@@ -153,56 +175,66 @@ contract AftermarketDevice is
             emit AftermarketDeviceNodeMinted(
                 newTokenId,
                 deviceAddress,
-                msg.sender
+                manufacturerNodeOwner
             );
         }
 
-        // Validate request and transfer funds to foundation
-        // This transfer is at the end of the function to prevent reentrancy
-        _validateMintRequest(msg.sender, devicesAmount);
+        // Validate license and transfer funds to foundation
+        _validateMintRequest(manufacturerNodeOwner, msg.sender, devicesAmount);
     }
 
-    /// @notice Claims the ownership of a list of aftermarket devices to a list of owners
-    /// @dev Caller must have the admin role
-    /// @dev This contract must be approved to spend the tokens in advance
-    /// @param adOwnerPair List of pairs AD-owner
+    /**
+     * @notice Claims the ownership of a list of aftermarket devices to a list of owners
+     * @dev Caller must have the admin role
+     * @dev This contract must be approved to spend the tokens in advance
+     * @param adOwnerPair List of pairs AD-owner
+     *  aftermarketDeviceNodeId -> Token ID of the AD
+     *  owner -> Address to be the new AD owner
+     */
     function claimAftermarketDeviceBatch(
         AftermarketDeviceOwnerPair[] calldata adOwnerPair
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        AftermarketDeviceStorage.Storage storage ads = AftermarketDeviceStorage
-            .getStorage();
-        INFT adIdProxy = INFT(ads.idProxyAddress);
-
-        uint256 aftermarketDeviceNode;
-        address owner;
-        for (uint256 i = 0; i < adOwnerPair.length; i++) {
-            aftermarketDeviceNode = adOwnerPair[i].aftermarketDeviceNodeId;
-            owner = adOwnerPair[i].owner;
+        _claimAftermarketDeviceBatch(adOwnerPair);
+    }
 
-            require(
-                !ads.deviceClaimed[aftermarketDeviceNode],
-                "Device already claimed"
-            );
+    /**
+     * @notice Claims the ownership of a list of aftermarket devices to a list of owners
+     * The manufacturer node owner must grant the claimer privilege to the caller
+     * @dev This contract must be approved to spend the tokens in advance
+     * @param adOwnerPair List of pairs AD-owner
+     *  aftermarketDeviceNodeId -> Token ID of the AD
+     *  owner -> Address to be the new AD owner
+     */
+    function claimAftermarketDeviceBatchAuthorized(
```

Comment:
Can I ask the same question here—can we consolidate?

Reply:
That is true. I was a little skeptical about that due to the admin role required in this case, but I think might be worth too.

Reply:
Oh, that's a good point. I'm not sure then.